padertorch/modules/normalization.py

```diff
@@ -1,104 +1,337 @@
+                        y_ = y_ / torch.sqrt(self.runnning_var.data)
+                    y = y + self.interpolation_factor * (y_ - y).detach()
+                    y = y * compute_mask(y, sequence_lengths, self.batch_axis, self.sequence_axis)
         else:
-            raise ValueError(f'chosen order {self.order} in is not'
-                             f' known in {self}')
+            y = x
+            if self.shift:
+                y = y - self.running_mean.data
+            if self.scale:
+                y = y / torch.sqrt(self.runnning_var.data)
+            if self.gamma is not None:
+                y = y * self.gamma
+            if self.beta is not None:
+                y = y + self.beta
+            y = y * compute_mask(x, sequence_lengths, self.batch_axis, self.sequence_axis)
+        return y
+
+    def inverse(self, x):
+        if not self.track_running_stats:
+            raise NotImplementedError
+        if self.beta is not None:
+            x = x - self.beta
+        if self.gamma is not None:
+            x = x / self.gamma
+        if self.scale:
+            x = torch.sqrt(self.running_var) * x
+        if self.shift:
+            x = x + self.running_mean
+        return x
+
+
+class Normalize(Function):
+    @staticmethod
+    def forward(ctx, x, gamma, beta, statistics_axis, batch_axis, sequence_axis, sequence_lengths, shift, scale, eps):
+        ctx.statistics_axis = statistics_axis
+        ctx.batch_axis = batch_axis
+        ctx.sequence_axis = sequence_axis
+        ctx.seq_len = sequence_lengths
+        ctx.shift = shift
+        ctx.scale = scale
+        ctx.eps = eps
 
-        norm += self.norm_epsilon
-        if self.order == 'l2':
-            norm = pt.ops.pointwise.sqrt(norm)
-        if self.order == 'mean':
-            norm = None
+        # compute mask
+        mask = compute_mask(x, sequence_lengths, batch_axis, sequence_axis)
 
-        return mean, norm
+        # compute statistics
+        n_values = mask.sum(dim=statistics_axis, keepdim=True)
+        x = x * mask
+        mean = x.sum(dim=statistics_axis, keepdim=True) / torch.max(n_values, torch.ones_like(n_values))
+        power = (x ** 2).sum(dim=statistics_axis, keepdim=True) / torch.max(n_values, torch.ones_like(n_values))
+        y = x
+        if shift:
+            y = y - mean
+            power = power - mean**2
+        if scale:
+            y = y / torch.sqrt(power + eps)
+        ctx.save_for_backward(x, gamma, beta, mean, power)
 
-    def forward(self, tensor):
-        if isinstance(tensor, PackedSequence):
+        if gamma is not None:
+            assert gamma.dim() == x.dim(), gamma.shape
+            y = y * gamma
+        if beta is not None:
+            assert beta.dim() == x.dim(), beta.shape
+            y = y + beta
+        return y*mask, mean, power, n_values
+
+    @staticmethod
+    def backward(ctx, grad_y, grad_mean, grad_power, _):
+        # equations from https://arxiv.org/abs/1502.03167
```

Comment:
You should add a small doctext expalining why you are overwriting the backward step